addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/ForbiddenBypassScanRule.java

```diff
@@ -0,0 +1,194 @@
+        return 40038;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+    }
+
+    @Override
+    public void scan() {
+        HttpMessage message = getNewMsg();
+        try {
+            sendAndReceive(message);
+            if (message.getResponseHeader().getStatusCode() == HttpStatusCode.FORBIDDEN) {
+                URI uri = message.getRequestHeader().getURI();
+                String host = uri.getAuthority();
+                String path = uri.getPath();
+                String schema = uri.getScheme();
+                sendPathPayloads(path, host, schema, uri);
+                sendHeaderPayloads(path, host, schema, uri); 
+            }
+        } catch (IOException e) {
+            LOG.error(e.getMessage(), e);
+        }
+    }
+    
+    public void sendPathPayloads(String path, String host, String schema, URI uri) throws IOException {
+        String[] pathPayloads =
+                {
+                        "/%2e" + path,
+                        path + "/./",
+                        "/." + path + "/./",
+                        path + "%20/",
+                        "/%20" + path + "%20/",
+                        path + "..;/",
+                        path + "?",
+                        path + "??",
+                        "/" + path + "//",
+                        path + "/",
+                        path + "/.testus",
+                        path + "../app.py"
+                };
+
+        for (String pathPayload : pathPayloads) {
+            String tmpUri = schema + "://" + host + "" + pathPayload;
+            HttpMessage reqWithPayload =
+                    new HttpMessage(new org.apache.commons.httpclient.URI(tmpUri, true));
+            sendAndReceive(reqWithPayload);
+
+            if (reqWithPayload.getResponseHeader().getStatusCode() == HttpStatusCode.OK) {
+                alertFound(uri, reqWithPayload);
+            }
+        }
+    }
+    
+    public void sendHeaderPayloads(String path, String host, String schema, URI uri) throws IOException {
+        String[] headerPayloads =
+                {
+                        "X-Rewrite-URL: " + path,
+                        "X-Original-URL: " + path,
+                        "Referer: " + path,
+                        "X-Custom-IP-Authorization: 127.0.0.1",
+                        "X-Originating-IP: 127.0.0.1",
+                        "X-Forwarded-For: 127.0.0.1",
+                        "X-Remote-IP: 127.0.0.1",
+                        "X-Client-IP: 127.0.0.1",
+                        "X-Host: 127.0.0.1",
+                        "X-Forwared-Host: 127.0.0.1"
+                };
+
+        for (String header : headerPayloads) {
+            String tmpUri = schema + "://" + host;
+            String[] headerPayload = header.split(":");
```

Comment:
Can be declared later.

Reply:
Hi what can be declared later? headerPayload?